### ISSUE TITLE
Element zapper or picker don't appear properly in Firefox on https://www.mnot.net/blog/

### ISSUE THREAD

**Issue by siebenmann** [uBlockOrigin/uBlock-issues#170]
### Prerequisites

- [x] I verified that this is not a filter issue
    - Filter issues MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues)
    
- [x] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uBlockOrigin](https://old.reddit.com/r/uBlockOrigin/)
    
- [x] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported.
- I tried to reproduce the issue when...
    - [x] uBlock Origin is the only extension
    - [x] uBlock Origin with default lists/settings
    - [x] using a new, unmodified browser profile
- [x] I am running the latest version of uBlock Origin
- [x] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### Description

If I try to invoke the element zapper or element picker on https://www.mnot.net/blog/ or one of its article pages, such as https://www.mnot.net/blog/2018/08/14/trust_australia, they don't really work. Instead, something that looks like an unstyled and mostly blank version of the element picker UI appears in the top left corner of the page. This may be due to an interaction with the page's Content Security Policy, because the browser console reports:

````
Content Security Policy: The page’s settings blocked the loading of a resource at self (“style-src”).
Source: html#ublock0-epicker, #ublock0-epicker ....
````

### A specific URL where the issue occurs

https://www.mnot.net/blog/

### Steps to Reproduce

1. Go to  https://www.mnot.net/blog/
2. Invoke uBlock Origin's element zapper
3. Observe things not working

### Expected behavior:

The usual element zapper UI appears and works; no element picker UI visible, most parts of the page grayed out, and the element I'm hovering the mouse cursor over is highlighted.

### Actual behavior:

A mangled version of the element picker UI appears in the top left corner, no highlighting of anything happens, clicking the mouse in most of the page does nothing, and clicking the mouse in the mangled UI makes it disappear.

### Your environment

* uBlock Origin version: this happens in both 1.16.16 and 1.16.17b4
* Browser Name and version: Firefox 61 (and it also happens in a recent version of Firefox Nightly)
* Operating System and version: Fedora Linux 28 (64-bit)

I've tested this in a completely new browser profile with a completely new and un-customized install of 1.16.17b4 and had it happen.

**Comment by gorhill** [comment]
Duplicate of <https://github.com/gorhill/uBlock/issues/2240>.

**Comment by gwarser** [comment]
I thought this was fixed https://bugzilla.mozilla.org/show_bug.cgi?id=1415352

**Comment by gorhill** [comment]
I need to investigate why this still happens if this was fixed in Firefox.

**Comment by gorhill** [comment]
I do get the following in the console:

>     Content Security Policy: The page’s settings blocked the loading of a resource at self (“style-src”). >     Source: 
>     html#ublock0-epicker,
>     #ublock0-epicker ....

So it does not look like that Firefox fix applies in the current case. It looks to me that the style being forbidden is the one inside uBO's iframe

**Comment by gorhill** [comment]
I tried to experiment with user stylesheet but it turns out using user stylesheet for that case is anything but trivial, due to the fact that the stylesheet needs to be in an iframe for which uBO does not have a frame id.

Then I thought that maybe removing the style tag before inserting the element picker DOM in the frame and reinserting manually could work, but it didn't.

In a last attempt instead of removing/reinserting the style tag, I created a new one and filled it with the old style tag content, and to my surprise this worked. So apparently the Firefox quick applies to style tag directly created from within a content script.
